src/index.js

```diff
@@ -18,6 +26,20 @@ function convert (json, options = {}) {
     return {};
   }
 
+  // check for a URL, before testing against the type "string",
+  // because every URL is also a string 
+  if(regex.url.test(json)){
+    return { type: 'url'};
```

Comment:
All of these tests should go after ` if (typeof json === 'string') {` and when the if condition goes through it should return `{type: 'string', format: 'url'}`.